backend/src/database/users.go

```diff
@@ -307,3 +333,83 @@ func (db *DB) GetLoginActivity(days int, facilityID *uint) ([]models.LoginActivi
 	}
 	return acitvity, nil
 }
+
+type LoginEngagementActivityWithUserInfo struct {
+	UserInfo            models.User                `json:"user_info"`
+	UserEngagementTimes []models.SessionEngagement `json:"user_engagement_times"`
+}
+
+func (db *DB) GetLoginEngagementActivity(userID int) (*LoginEngagementActivityWithUserInfo, error) {
+	var sessionEngagement []models.SessionEngagement
+
+	query := db.Table("user_session_tracking as ust").
+		Select(`ust.user_id,
+	ANY_VALUE(u.name_first) AS name_first,
+	ANY_VALUE(u.name_last) AS name_last,
+	ANY_VALUE(u.username) AS username,
+	TO_CHAR(DATE(ust.session_start_ts), 'YYYY-MM-DD"T"HH24:MI:SS"Z"') AS time_interval,
+	SUM(EXTRACT(EPOCH FROM ust.session_duration) / 3600) AS total_hours,
+	SUM(EXTRACT(EPOCH FROM ust.session_duration) / 60) AS total_minutes`).
+		Joins("JOIN users u ON ust.user_id = u.id").
+		Where("ust.user_id = ?", userID).
+		Group("ust.user_id, time_interval").
+		Order("ust.user_id, time_interval")
```

Comment:
@corypride @carddev81 Should this be filtered to only include the last 30 days? Or are we needing to pull in all of the user_session_tracking data?

Reply:
I'm going to decide yes until I hear otherwise.